### PR TITLE
feat: primitive set formula

### PR DESCRIPTION
**Pull request by JonathanStarup**
Related to #5229 

Add simple SetFormula objects based on BoolFormula. Vars are simple integers and constants are sets of integers as well using the same mechanism. This should make thing easy in the set formula world and keep the translation simple too.

### CODE REVIEW

**Comment by JonathanStarup** [comment]
@magnus-madsen is this what you had in mind?

File: main/src/ca/uwaterloo/flix/language/phase/unification/SetFormula.scala [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import scala.collection.immutable.SortedSet
+
+sealed trait SetFormula {
+
+  /**
+    * Returns the free variables in `this` expression.
+    */
+  final def freeVars: SortedSet[Int] = this match {
+    case SetFormula.All => SortedSet.empty
+    case SetFormula.Constant(_) => SortedSet.empty
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/SetFormula.scala:16]: Constant -> Cst (following similar naming).

**Comment by magnus-madsen** [comment]
Yes, plus some smart constructors.

**Comment by magnus-madsen** [comment]
We can refine those smart constructors later.

**Comment by JonathanStarup** [comment]
I added some, Should we stick with boolean language? Change it to set terminology? or maybe both?

**Comment by magnus-madsen** [comment]
I merged it, but I don't think we need the Alg part-- at least for now.

**Pull request merged**